src/PyNvCodec/src/PyNvEncoder.cpp

```diff
@@ -75,8 +75,8 @@ std::map<NV_ENC_CAPS, int> PyNvEncoder::Capabilities()
 bool PyNvEncoder::Reconfigure(const map<string, string>& encodeOptions,
                               bool force_idr, bool reset_enc, bool verbose)
 {
+  NvEncoderClInterface cli_interface(encodeOptions);
```

Comment:
fail as early as possible. on construction, when the user set the wrong parameters. No just when encoder is set up.